db/keyval/redis/plugin_impl.go

```diff
@@ -16,63 +16,65 @@ package redis
 
 import (
 	"github.com/ligato/cn-infra/core"
+	"github.com/ligato/cn-infra/datasync"
+	"github.com/ligato/cn-infra/datasync/persisted/dbsync"
 	"github.com/ligato/cn-infra/db/keyval/plugin"
 	"github.com/ligato/cn-infra/logging"
 	"github.com/ligato/cn-infra/servicelabel"
-	"github.com/namsral/flag"
 )
 
 // PluginID used in the Agent Core flavors
 const PluginID core.PluginName = "RedisClient"
 
 // Plugin implements Plugin interface therefore can be loaded with other plugins
 type Plugin struct {
-	LogFactory     logging.LogFactory
-	ServiceLabel   *servicelabel.Plugin
-	ConfigFileName string
+	Transport    datasync.TransportAdapter
```

Comment:
Same as for etcdv3 plugin. We just need one constructor that takes keyval/plugin_api.go argument to initialize transport independently from a particular keyval/*/plugin_impl.go